examples/user_info_custom_provider/src/oauth2/slack.rs

```diff
@@ -0,0 +1,27 @@
+use rocket::fairing::Fairing;
+use rocket::request::Request;
+use rocket::response::Redirect;
+use rocket_oauth2::hyper_sync_rustls_adapter::HyperSyncRustlsAdapter;
+use rocket_oauth2::{OAuth2, TokenResponse};
+
+pub fn fairing() -> impl Fairing {
+    OAuth2::fairing(
+        HyperSyncRustlsAdapter,
+        post_install_callback,
+        "slack",
+        "/oauth2/slack",
+        Some(("/login/slack", vec![String::from("identity:basic")])),
```

Comment:
This code currently isn't working. I think due to issues with how Slack implements OAuth scopes. 

It looks like the URL generated by `rocket_oauth2` does not match what Slack expects. I'm not sure if this is an issue with the Slack implementation of OAuth2 or something that could be fixed in `rocket_oauth2`? If you could help me understand that I'd greatly appreciate it!

## Reproduction Notes

There is a distinction in Slack between "bot" and "user" scopes.

![Screenshot from 2020-02-16 13 29 54](https://user-images.githubusercontent.com/3886290/74613125-7b40ab00-50c0-11ea-816a-aa843d426186.png)

When I visit `http://localhost:8000/login/slack` I get redirected to `https://international-tech.slack.com/oauth?client_id=498993223299.941527274290&redirect_uri=http%3A%2F%2Flocalhost%3A8000%2Foauth2%2Fslack&state=ZGtxJ_xLXq6pbKHwlW9EnQ&scope=identity.basic&user_scope=&granular_bot_scope=1&team=&install_redirect=&single_channel=0&tracked=1`

![Screenshot from 2020-02-16 13 36 09](https://user-images.githubusercontent.com/3886290/74613293-ceffc400-50c1-11ea-97ae-84aa8ea11aaa.png)


However by editing the Slack authorization URL to remove the `scope` parameter and set the value of `user_scope` to be `identity.basic`
```diff
-state=ZGtxJ_xLXq6pbKHwlW9EnQ&scope=identity.basic&user_scope=&granular_bot_scope=1
+ state=ZGtxJ_xLXq6pbKHwlW9EnQ&user_scope=identity.basic&granular_bot_scope=1
```

Result: `https://international-tech.slack.com/oauth?client_id=498993223299.941527274290&redirect_uri=http%3A%2F%2Flocalhost%3A8000%2Foauth2%2Fslack&state=ZGtxJ_xLXq6pbKHwlW9EnQ&scope=identity.basic&user_scope=&granular_bot_scope=1&team=&install_redirect=&single_channel=0&tracked=1`

![Screenshot from 2020-02-16 13 36 46](https://user-images.githubusercontent.com/3886290/74613289-c7d8b600-50c1-11ea-977e-217fa8ac4830.png)

Unfortunately this doesn't seem to work either. Resulting in `Error: Token exchange failed: Error { kind: ExchangeFailure, source: Some("TokenResponse access_token was missing or not a string") }`

Reply:
If you have any guidance on how to deal with the issue above, I'd appreciate it! The same approach seems to work fine when only requesting Bot scopes. Which leads me to think the problem is related to Slack's implementation.

![Screenshot from 2020-02-16 13 50 59](https://user-images.githubusercontent.com/3886290/74613455-6f0a1d00-50c3-11ea-8c4e-66d93e21c6c1.png)